src/Compilers/Test/Core/CompilationVerifier.cs

```diff
@@ -220,6 +227,132 @@ public void Emit(string expectedOutput, int? expectedReturnCode, string[] args,
             }
         }
 
+        private class Resolver : ILVerify.ResolverBase
+        {
+            private readonly Dictionary<string, ImmutableArray<byte>> imagesByName = new Dictionary<string, ImmutableArray<byte>>();
+
+            internal Resolver(IList<ModuleData> allModuleData)
+            {
+                foreach (var module in allModuleData)
+                {
+                    string name = module.SimpleName;
+                    if (imagesByName.ContainsKey(module.SimpleName))
+                    {
+                        throw new Exception($"Multiple modules named '{name}' were found");
+                    }
+                    imagesByName.Add(name, module.Image);
+                }
+            }
+
+            protected override PEReader ResolveCore(string simpleName)
+            {
+                if (imagesByName.TryGetValue(simpleName, out var image))
+                {
+                    if (image.IsDefault)
+                    {
+                        throw new Exception($"ILVerify was not able to resolve a module named '{simpleName}' because multiple exist in this compilation");
+                    }
+                    return new PEReader(image);
+                }
+
+                throw new Exception($"ILVerify was not able to resolve a module named '{simpleName}'");
+            }
+        }
+
+        private void ILVerify(Verification verification)
+        {
+            if (verification == Verification.Skipped)
+            {
+                return;
+            }
+
+            try
+            {
+                var resolver = new Resolver(_allModuleData);
```

Comment:
Note to self: Add a note/issue to get the module bits without loading them in memory